src/Explorer/Controls/Settings/SettingsUtils.tsx

```diff
@@ -0,0 +1,331 @@
+import * as React from "react";
+import * as AutoPilotUtils from "../../../Utils/AutoPilotUtils";
+import * as SharedConstants from "../../../Shared/Constants";
+import * as CommonConstants from "../../../Common/Constants";
+import * as DataModels from "../../../Contracts/DataModels";
+import { AutopilotTier } from "../../../Contracts/DataModels";
+import {
+  computeAutoscaleUsagePriceHourly,
+  getPriceCurrency,
+  getCurrencySign,
+  getAutoscalePricePerRu,
+  getMultimasterMultiplier,
+  computeRUUsagePriceHourly,
+  getPricePerRu,
+  calculateEstimateNumber
+} from "../../../Utils/PricingUtils";
+
+/*
+Class used to track baseline and current values of a state object. If both the values differ,
+isDirty is set to true. The isDirty value is used to change css of the component and to govern logic
+around whether the current value should be persistently saved or not.
+*/
+export type StatefulValuesType = boolean | string | number | DataModels.IndexingPolicy;
+
+export interface StatefulValue<StatefulValuesType> {
+  baseline: StatefulValuesType;
+  current: StatefulValuesType;
+  isValid: boolean;
+}
+
+const getStringValue = (value: StatefulValuesType): string => {
+  const type = typeof value;
+  switch (type) {
+    case "string":
+    case "undefined":
```

Comment:
In the "undefined" type, it would return "undefined". Is it what we want?

Reply:
yes, it would still lead to an equivalent comparison.